src/template_parser.rs

```diff
@@ -1088,37 +1100,66 @@ fn build_expression<'a, C: 'a>(
         ExpressionKind::List(nodes) => {
             let templates = nodes
                 .iter()
-                .map(|node| build_expression(node, build_keyword).map(|x| x.into_template()))
+                .map(|node| build_expression(language, node).map(|x| x.into_template()))
                 .try_collect()?;
             Ok(Expression::Template(Box::new(ListTemplate(templates))))
         }
-        ExpressionKind::FunctionCall(function) => build_global_function(function, build_keyword),
-        ExpressionKind::MethodCall(method) => build_method_call(method, build_keyword),
-        ExpressionKind::AliasExpanded(id, subst) => build_expression(subst, build_keyword)
-            .map_err(|e| e.within_alias_expansion(*id, node.span)),
+        ExpressionKind::FunctionCall(function) => build_global_function(language, function),
+        ExpressionKind::MethodCall(method) => build_method_call(language, method),
+        ExpressionKind::AliasExpanded(id, subst) => {
+            build_expression(language, subst).map_err(|e| e.within_alias_expansion(*id, node.span))
+        }
+    }
+}
+
+struct CommitTemplateLanguage<'a, 'b> {
```

Comment:
Yeah,`'repo` makes sense here. I wouldn't give the other locals an explicit name because we don't have to care. It's also fine to clone WorkspaceId. It's just not required.

Reply:
I'll address this after splitting the modules. It's unclear for me at which commit I can rename a generic `'a` lifetime to `'repo`.